arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -879,61 +797,40 @@ bool SynchronizeShard::first() {
-             "'"
-          << database << "/" << shard << "' for central '" << database << "/"
-          << planId << "'";
-
-      // now do a final sync-to-disk call. note that this can fail
-      auto& engine = vocbase->server().getFeature<EngineSelectorFeature>().engine();
-      Result res = engine.flushWal(/*waitForSync*/ true, /*waitForCollector*/ false);
-      if (res.fail()) {
-        LOG_TOPIC("a49d1", INFO, Logger::MAINTENANCE) << res.errorMessage();
-        result(res);
-        return false;
-      }
-
-      try {
-        NetworkFeature& nf = _feature.server().getFeature<NetworkFeature>();
-        network::ConnectionPool* pool = nf.pool();
-        auto asResult = addShardFollower(pool, ep, database, shard, 0, clientId,
-                                         SyncerId{}, _clientInfoString, 60.0);
-
-        if (asResult.ok()) {
-          if (Logger::isEnabled(LogLevel::DEBUG, Logger::MAINTENANCE)) {
-            std::stringstream msg;
-            msg << "SynchronizeShard: shortcut worked, done, ";
-            AppendShardInformationToMessage(database, shard, planId, startTime, msg);
-            LOG_TOPIC("f4a5b", DEBUG, Logger::MAINTENANCE) << msg.str();
-          }
-          collection->followers()->setTheLeader(leader);
-          return false;
-        }
-        if (asResult.errorNumber() != TRI_ERROR_REPLICATION_SHARD_NONEMPTY) {
-          // Stop action in this case
-          LOG_TOPIC("daaaa", INFO, Logger::MAINTENANCE) << "SynchronizeShard, error in addFollower (short cut): " << asResult.errorMessage();
-          result(asResult);
-          return false;
-        }
-        // Otherwise move on.
-      } catch (...) {
-      }
-    }
-#endif
-
     LOG_TOPIC("53337", DEBUG, Logger::MAINTENANCE)
         << "synchronizeOneShard: trying to synchronize local shard '" << database
         << "/" << shard << "' for central '" << database << "/" << planId << "'";
 
+    std::shared_ptr<DatabaseTailingSyncer> tailingSyncer;
+    VPackBuilder builder;
+    // build configuration for WAL tailing
+    {
+      {
+        VPackObjectBuilder o(&builder);
+        builder.add(ENDPOINT, VPackValue(ep));
+        builder.add(DATABASE, VPackValue(getDatabase()));
+        builder.add(COLLECTION, VPackValue(getShard()));
+        builder.add(LEADER_ID, VPackValue(leader));
+        builder.add("requestTimeout", VPackValue(600.0));
+        builder.add("connectTimeout", VPackValue(30.0));
+      }
+      
+      ReplicationApplierConfiguration configuration =
+          ReplicationApplierConfiguration::fromVelocyPack(feature().server(), builder.slice(), getDatabase());
+      // will throw if invalid
+      configuration.validate();
+  
+      // build DatabaseTailingSyncer object for WAL tailing
+      TRI_ASSERT(tailingSyncer == nullptr);
+      tailingSyncer = DatabaseTailingSyncer::create(guard.database(), configuration, /*lastTick*/ 0, /*useTick*/true);
```

Comment:
I suggest to check `tailingSyncer` for `nullptr` here such that we know later on that it is valid. This saves several checks later. And if it is `nullptr` here, we might as well give up right away.